src/cli.js

```diff
@@ -0,0 +1,59 @@
+const program = require('commander');
+const SemanticReleaseError = require('@semantic-release/error');
+const logger = require('./lib/logger');
+
+function list(values) {
+  return values.split(',').map(value => value.trim());
+}
+
+module.exports = async () => {
+  program
+    .name('semantic-release')
+    .description('Run automated package publishing')
+    .option('-b, --branch <branch>', 'Branch to release from')
+    .option('--github-token <token>', 'Token to authenticate with Github API')
+    .option('--github-url <url>', 'GitHub Enterprise endpoint')
+    .option('--github-api-path-prefix <prefix>', 'Prefix of the GitHub Enterprise endpoint')
+    .option(
+      '--verify-conditions <paths>',
+      'Comma separated list of paths or packages name for the verifyConditions plugin',
+      list
+    )
+    .option('--get-last-release <path>', 'Path or package name for the getLastRelease plugin')
+    .option('--analyze-commits <path>', 'Path or package name for the analyzeCommits plugin')
+    .option(
+      '--verify-release <paths>',
+      'Comma separated list of paths or packages name for the verifyRelease plugin',
+      list
+    )
+    .option('--generate-notes <path>', 'Path or package name for the generateNotes plugin')
+    .option('--debug', 'Output debugging information')
+    .option(
+      '-d, --dry-run',
+      'Dry-run mode, skipping verifyConditions, publishing and release, printing next version and release notes'
+    )
+    .parse(process.argv);
+
+  if (program.debug) {
+    // Debug must be enabled before other requires in order to work
+    require('debug').enable('semantic-release:*');
+  }
+
+  try {
+    if (program.args.length > 0) {
+      program.outputHelp();
+      process.exitCode = 1;
+    } else {
+      await require('./index')(program);
+    }
+  } catch (err) {
+    // If error is a SemanticReleaseError then it's an expected exception case (no release to be done, running on a PR etc..) and the cli will return with 0
+    // Otherwise it's an unexpected error (configuration issue, code issue, plugin issue etc...) and the cli will return 1
+    if (err instanceof SemanticReleaseError) {
```

Comment:
Could a plugin have a different version of `@semantic-release/error` that makes this check fail? Would checking `err.name` work too?

Reply:
Could you clarify what do you mean by "a different version of @semantic-release/error" ?

The idea here is that throwing a `@semantic-release/error` will stop the process and exit gracefully without an error.
While throwing any other error is considered and actual problem that prevent `semantic-release` to work and will result in logging the error with a stack trace and exiting with 1.

Reply:
If a plugin has a dependency on `@semantic-releae/error` `^1.0.0` and this module has a dependency on `@semantic-release/error` `^2.0.0` the `instanceof` expression will always be false because it will be different classes, so this logic depends on the npm deduplication logic.

Reply:
Yes indeed. I didn't think about this case.
I realize there is a problem in the definition of @semantic-release/error as the name is not set properly.
```js
const SemanticReleaseError = require('@semantic-release/error');

const err = new SemanticReleaseError('message', 'CODE');
console.log(err.name);
// Error
```
Ideally we need to fix that and use `err.name` in this test.